Adjusting speak and listen css so buttons fit screen

raivisdejus opened the pull request:
## Pull Request Form

### Type of Pull Request

- [x] Related to a listed issue 
* https://github.com/common-voice/common-voice/issues/3969

This PR fixes the issue where on zoomed screens some buttons do not fit the screen and as zoom is disabled on the contribution pages buttons are unavailable

Changes:
* enabled scroll on `/speak` and `/listen` if contents do not fit the screen (removed `overlay:hidden;`)
* set height of `/speak` and `/listen` to `100dvh - 2px` so it fits mobile screens snugly
* adjusted margins and paddings so the content fits better into the screen
* set min-height to 70%, so content is more compact and does not create a scroll
* ~adjusted `contribution.tsx` to not print primary buttons on `isDone` overlay screen, so the screen is smaller and visible parts fit the screen~

Tested on:
* Firefox, Chrome, Edge, Safari on mobiles, tablets and desktops (Linux, Mac, Windows)
* Zoom of 125% still fits without any scrolls, for larger zooms scroll appears but site is functional  

Screenshots:
* https://i.imgur.com/GehLhd8.png
* https://i.imgur.com/KYDYMNN.png
* https://i.imgur.com/HhaG0sO.png
* https://i.imgur.com/Z1mjLiP.png
* https://i.imgur.com/QD5neek.png
* https://i.imgur.com/kD8rkVE.png
* https://i.imgur.com/PbIPuZS.png

moz-rotimib commented:
Hello @raivisdejus thanks for opening this PR. I will take a look and let you know if I have any comments

raivisdejus commented:
@moz-rotimib Thanks for the comments. I have adjusted the PR in the following ways:
1. I reverted the change for primary buttons for the `isDone` state, it is not needed. Guest user `accept-privacy-and-terms` checkbox fits nicely on the screen even on small mobile devices.
2. Adjusted the `height` of the `.contribution` container so it matches closer to what is currently in production. https://i.imgur.com/hcwwhDk.png

If we want to keep spacing between the buttons close to what is currently in production, I do not see an easy way to do that and also keep the layout flexible enough to fit smaller screens or larger zoom levels. Below are a few screenshots on different screensizes.
* https://i.imgur.com/JabjFpV.png
* https://i.imgur.com/v7Suj9r.png
* https://i.imgur.com/OY6ux3Q.png
* https://i.imgur.com/Ev8y7Yw.png
* https://i.imgur.com/tMfozd4.png
* https://i.imgur.com/vaXZLGv.png

Will be happy to address further possible comments, so we get to the best result for all contributors with different screen size and zoom level combinations

moz-rotimib commented:
Hello, @raivisdejus  Thanks for the updates on this PR. 

I have made some changes to the contribution.css file which should help with the spacing of the buttons so it's closer to what is on production. Feel free to take a look and modify it if it can still be improved. Thanks a lot ❤️ 

[contribution.css.zip](https://github.com/common-voice/common-voice/files/11332268/contribution.css.zip)

raivisdejus commented:
@moz-rotimib I have added your changes to the PR and adjusted padding for `primary-buttons`:
1. to be `10vh` as in your changes if the `@media (-webkit-device-pixel-ratio: 1)` and `@media (-webkit-device-pixel-ratio: 2)`, this will cover most regular cases 
* https://i.imgur.com/rCimRNJ.png
* https://i.imgur.com/0XpL7Do.png
2. `6vh` by default to cover different zoom levels, otherwise on zooms 120-125% we get scroll https://i.imgur.com/aVVmllo.png

With latest adjustments:
* https://i.imgur.com/jG9WZpz.png
* https://i.imgur.com/3vS1VoC.png

moz-rotimib commented:
Thanks for the update @raivisdejus I'll take a look and let you if I have any comments

moz-rotimib commented on web/src/components/pages/contribution/contribution.css:
```diff
@@ -1,9 +1,5 @@
-.contribution-wrapper {
-    overflow: hidden;
-}
-
```
I think we should replace this with 

```overflow-x: hidden```

I was able to scroll horizontally on the speak/listen pages after recording/validating some clips

https://user-images.githubusercontent.com/113194143/235129495-d984b333-34ee-4002-8d4a-fac7b337bb4d.mov

raivisdejus replied:
Restored, safer that way indeed :)

moz-rotimib replied:
I'm unsure if you can reproduce this issue but with `overflow: hidden` I cannot scroll fully when the screen is resized (check the attached video). This is fixed when I use `overflow-x: hidden`. Do you notice the same behaviour?

https://user-images.githubusercontent.com/113194143/235159398-0ed85299-45bd-4274-9d4d-61d231b0d23d.mov

raivisdejus commented:
Issue with buttons not being visible on screens with small height is fixed. Large screens are also ok. And there is no scroll on tablets with small height in landscape.

Let's see if we got all the issues this time :)

moz-rotimib commented:
I will take a look and let you know if I have any comments. Thanks for the contribution ❤️

moz-rotimib commented on web/src/components/pages/contribution/contribution.css:
```diff
@@ -357,13 +353,13 @@
 
-        @media (--xl-up) {
-            margin-top: 12vh;
+        @media (min-height: 910px) {
```
I suggest we create a custom media query for this as we do in [media.css](https://github.com/common-voice/common-voice/blob/main/web/src/components/media.css) so we can keep track of the media queries we use in the application.

We can use the name - `xl-tall` for this value

moz-rotimib reviewed (approved):
I'm happy with the changes. Once the comment about creating a name for the media query in contribution.css, this can be merged.

Thanks once again for the contribution ❤️
